ament_flake8/ament_flake8/main.py

```diff
@@ -0,0 +1,264 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from distutils.version import LooseVersion
+import os
+import sys
+import time
+from xml.sax.saxutils import escape
+from xml.sax.saxutils import quoteattr
+
+import flake8
+if LooseVersion(flake8.__version__) >= '3.0':
+    from flake8.main import application as flake8_app
+    from flake8.api.legacy import StyleGuide
+
+
+def main(argv=sys.argv[1:]):
+    config_file = os.path.join(
+        os.path.dirname(__file__), 'configuration', 'ament_flake8.ini')
+
+    parser = argparse.ArgumentParser(
+        description='Check code using flake8.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--config',
+        metavar='path',
+        default=config_file,
+        dest='config_file',
+        help='The config file')
+    parser.add_argument(
+        '--linelength', metavar='N', type=int,
+        help='The maximum line length (default: specified in the config file)')
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help="The files or directories to check. For directories files ending "
+             "in '.py' will be considered.")
+    parser.add_argument(
+        '--exclude',
+        metavar='filename',
+        nargs='*',
+        dest='excludes',
+        help='The filenames to exclude.')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    args = parser.parse_args(argv)
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    if not os.path.exists(args.config_file):
+        print("Could not find config file '%s'" % args.config_file, file=sys.stderr)
+        return 1
+
+    report = generate_flake8_report(
+        args.config_file, args.paths, args.excludes,
+        max_line_length=args.linelength)
+
+    # print statistics about errors
+    if report.total_errors:
+        print('')
+        report.print_statistics()
+
+    # print summary
+    print('')
+    print('%d files checked' % len(report.files))
+    if not report.total_errors:
+        print('No errors or warnings')
+        rc = 0
+    else:
+        print('%d errors' % (report.total_errors))
+
+        print('')
+        error_type_counts = report.get_error_type_counts()
```

Comment:
This line fails when the legacy API is being used since [this class](https://github.com/ament/ament_lint/blob/16cb042e5d04e6d50b60f4c5d373d356052370d9/ament_flake8/ament_flake8/legacy.py#L27) doesn't have that method.